File: android/app/src/main/kotlin/com/ghoststream/vpn/ui/components/GhostBadge.kt
package com.ghoststream.vpn.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ghoststream.vpn.ui.theme.AccentPurple
import com.ghoststream.vpn.ui.theme.AccentTeal
import com.ghoststream.vpn.ui.theme.YellowWarning

enum class BadgeVariant { DEFAULT, ALT, WARN }

@Composable
fun GhostBadge(
    text: String,
    variant: BadgeVariant = BadgeVariant.DEFAULT,
    modifier: Modifier = Modifier,
) {
    val (fg, bg, border) = when (variant) {
        BadgeVariant.DEFAULT -> Triple(
            AccentPurple,
            AccentPurple.copy(alpha = 0.1f),
            AccentPurple.copy(alpha = 0.24f),
        )
        BadgeVariant.ALT -> Triple(
            AccentTeal,
            AccentTeal.copy(alpha = 0.1f),
            AccentTeal.copy(alpha = 0.24f),
        )
        BadgeVariant.WARN -> Triple(
            YellowWarning,
            YellowWarning.copy(alpha = 0.1f),
            YellowWarning.copy(alpha = 0.24f),
        )
    }

    Text(
        text = text,
        fontSize = 10.sp,
        fontWeight = FontWeight.Bold,
        letterSpacing = 0.3.sp,
        color = fg,
        modifier = modifier
            .clip(RoundedCornerShape(999.dp))
            .background(bg)
            .border(1.dp, border, RoundedCornerShape(999.dp))
            .padding(horizontal = 10.dp, vertical = 6.dp),
    )
}

File: ghoststream_flutter/lib/ui/widgets/ghost_badge.dart
import 'package:flutter/material.dart';

import '../../core/theme/ghost_colors.dart';

enum GhostBadgeVariant { accent, alt, warn }

class GhostBadge extends StatelessWidget {
  final String text;
  final GhostBadgeVariant variant;

  const GhostBadge({
    super.key,
    required this.text,
    this.variant = GhostBadgeVariant.accent,
  });

  @override
  Widget build(BuildContext context) {
    final gc = context.ghostColors;

    final (Color fg, Color bg, Color borderColor) = switch (variant) {
      GhostBadgeVariant.accent => (
          gc.accentPurple,
          gc.accentPurple.withValues(alpha: 0.1),
          gc.accentPurple.withValues(alpha: 0.24),
        ),
      GhostBadgeVariant.alt => (
          gc.accentTeal,
          gc.accentTeal.withValues(alpha: 0.1),
          gc.accentTeal.withValues(alpha: 0.24),
        ),
      GhostBadgeVariant.warn => (
          gc.yellowWarning,
          gc.yellowWarning.withValues(alpha: 0.1),
          gc.yellowWarning.withValues(alpha: 0.24),
        ),
    };

    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 6),
      decoration: BoxDecoration(
        color: bg,
        borderRadius: BorderRadius.circular(999),
        border: Border.all(color: borderColor),
      ),
      child: Text(
        text,
        style: TextStyle(
          fontSize: 10,
          fontWeight: FontWeight.w700,
          color: fg,
          letterSpacing: 0.3,
          height: 1,
        ),
      ),
    );
  }
}
